diff -burN freetype-2.5.0.1/builds/unix/install.mk freetype-2.5.0.1-PS3/builds/unix/install.mk
--- freetype-2.5.0.1/builds/unix/install.mk	2013-06-19 08:39:31.000000000 -0400
+++ freetype-2.5.0.1-PS3/builds/unix/install.mk	2014-07-05 16:16:46.274102713 -0400
@@ -43,10 +43,8 @@
           $(INSTALL_DATA)                                           \
             $$P $(DESTDIR)$(includedir)/freetype2/freetype/config ; \
         done
-	-$(DELETE) $(DESTDIR)$(includedir)/freetype2/freetype/cache/*
-	-$(DELDIR) $(DESTDIR)$(includedir)/freetype2/freetype/cache
-	-$(DELETE) $(DESTDIR)$(includedir)/freetype2/freetype/internal/*
-	-$(DELDIR) $(DESTDIR)$(includedir)/freetype2/freetype/internal
+	-$(DELETE) $(DESTDIR)$(includedir)/freetype2/freetype/cache
+	-$(DELETE) $(DESTDIR)$(includedir)/freetype2/freetype/internal
 	$(INSTALL_DATA) $(BUILD_DIR)/ft2unix.h \
           $(DESTDIR)$(includedir)/ft2build.h
 	$(INSTALL_DATA) $(OBJ_BUILD)/ftconfig.h                        \
