diff -burN NoRSX-0.3.1/NoRSX_Example/Makefile NoRSX-0.3.1-PS3/NoRSX_Example/Makefile
--- NoRSX-0.3.1/NoRSX_Example/Makefile	2013-07-09 12:15:21.000000000 -0400
+++ NoRSX-0.3.1-PS3/NoRSX_Example/Makefile	2014-07-04 12:02:30.033489512 -0400
@@ -39,7 +39,7 @@
 #---------------------------------------------------------------------------------
 # any extra libraries we wish to link with the project
 #---------------------------------------------------------------------------------
-LIBS	:=	-lNoRSX -lgcm_sys -lrsx -lsysutil -lio -lm -lfreetype -lz -lpixman-1 -lrt -llv2 -lsysmodule -lpng -lpngdec -ljpgdec
+LIBS	:=	-lNoRSX -lgcm_sys -lrsx -lsysutil -lio -lm -lfreetype -lpixman-1 -lrt -llv2 -lsysmodule -lpng -lz -lpngdec -ljpgdec
 
 #---------------------------------------------------------------------------------
 # list of directories containing libraries, this must be the top level containing
